instrumentation/opentelemetry-instrumentation-aiohttp-server/tests/test_aiohttp_server_integration.py

```diff
@@ -0,0 +1,123 @@
+from opentelemetry.semconv.trace import SpanAttributes
+from opentelemetry.util._importlib_metadata import entry_points
+
+from opentelemetry.test.globals_test import (
+    reset_trace_globals,
+)
+
+
+@pytest.fixture(scope="session")
+def tracer():
+    test_base = TestBase()
+
+    tracer_provider, memory_exporter = test_base.create_tracer_provider()
+
+    reset_trace_globals()
+    trace_api.set_tracer_provider(tracer_provider)
+
+    yield tracer_provider, memory_exporter
+
+    reset_trace_globals()
+
+
+async def default_handler(request, status=200):
+    return aiohttp.web.Response(status=status)
+
+
+@pytest_asyncio.fixture
+async def server_fixture(tracer, aiohttp_server):
+    _, memory_exporter = tracer
+
+    AioHttpServerInstrumentor().instrument()
+
+    app = aiohttp.web.Application()
+    app.add_routes(
+        [aiohttp.web.get("/test-path", default_handler)])
+
+    server = await aiohttp_server(app)
+    yield server, app
+
+    memory_exporter.clear()
+
+    AioHttpServerInstrumentor().uninstrument()
+
+
+def test_checking_instrumentor_pkg_installed():
+    itered_entry_points = iter(entry_points(
+        "opentelemetry_instrumentor", "aiohttp-server"))
+
+    instrumentor = next(itered_entry_points).load()()
+    assert (isinstance(instrumentor, AioHttpServerInstrumentor))
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("url, expected_method, expected_status_code", [
+    ("/test-path", HTTPMethod.GET, HTTPStatus.OK),
+    ("/not-found", HTTPMethod.GET, HTTPStatus.NOT_FOUND)
+])
+async def test_status_code_instrumentation(tracer, server_fixture,
+                                           aiohttp_client, url,
+                                           expected_method,
+                                           expected_status_code):
+    _, memory_exporter = tracer
+    server, app = server_fixture
+
+    assert len(memory_exporter.get_finished_spans()) == 0
+
+    client = await aiohttp_client(server)
+    await client.get(url)
+
+    assert len(memory_exporter.get_finished_spans()) == 1
+
+    [span] = memory_exporter.get_finished_spans()
+
+    assert expected_method == span.attributes[SpanAttributes.HTTP_METHOD]
+    assert expected_status_code == span.attributes[SpanAttributes.HTTP_STATUS_CODE]
+
+    assert f"http://{server.host}:{server.port}{url}" == span.attributes[
+        SpanAttributes.HTTP_URL
+    ]
+
+
+@pytest.mark.skip(reason="Historical purposes. Can't see the reason of this mock.")
+def test_not_recording(self):
+    mock_tracer = mock.Mock()
+    mock_span = mock.Mock()
+    mock_span.is_recording.return_value = False
+    mock_tracer.start_span.return_value = mock_span
+    with mock.patch("opentelemetry.trace.get_tracer") as patched:
+        patched.start_span.return_value = mock_span
+        # pylint: disable=W0612
+        # host, port = run_with_test_server(
+        #     self.get_default_request(), self.URL, self.default_handler
+        # )
```

Comment:
Is this intentional?

Reply:
I already skipped that test. Like, I couldn't saw the reason for the mock and the test itself.
If you guys can't see a reason for keeping it I could remove it.

Reply:
Wdym by historical, this is a brand new file. If you'd like to remove it I wouldn't be opposed.

I think what it's testing is that the instrumentation short circuits tracing if the span isn't recording, as an optimization.

Reply:
It's historical because this PR continues the works of a previous one from a different author. The test was already there when we continued it.

Like, the way it was mocked the test was always passing.

Reply:
Feel free to delete the test!

Reply:
Actually, please remove the test.

Reply:
Removed.